hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3AEncryptionDSSEKMSUserDefinedKeyWithEncryptionContext.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.s3a;
```

Comment:
your ITests are not very useful as they don't assert on anything. I also don't think you need 3 new ITests that extend AbstractTestS3AEncryption. 

I recommend that you add a single ITest, that writes an object with encryption context. parameterize it for SSE-KMS default key, SS-KMS customer key, DSSE-KMS. Then do a GET on the object and assert on the encryption context.

Reply:
I will create just a single ITest class. Regarding the assertion, I couldn't find any S3 API that allow me to get the object's encryption context, `HeadObject`, for example, returns only `ServerSideEncryption` and `SSEKMSKeyId`.

The way that I found to test it is add a condition to the IAM Role policy or KMS Key policy requiring the encryption context to be set, like:

```
  "Condition": {
    "StringEquals": {
      "kms:EncryptionContext:project": "hadoop",
      "kms:EncryptionContext:jira": "HADOOP-19197"
    }
  }
```

Let me know if you know a way to get the encryption context from S3 or if you have a better idea on how to test it.

Reply:
I executed `mvn -Dparallel-tests -DtestsThreadCount=16 clean verify` in us-east-1, and I updated the PR description adding this information and explained how I tested that the encryption context is set in the S3 objects.